src/besselk.jl

```diff
@@ -203,4 +258,125 @@ function besselk_large_orders_scaled(v, x::T) where T <: Union{Float32, Float64,
+# Inu and Inum1 are generated from the power series form where K_{nu_1}/K_{nu}
+# is calculated with continued fractions. 
+# The continued fraction K_{nu_1}/K_{nu} method is a slightly modified form
+# https://github.com/heltonmc/Bessels.jl/issues/17#issuecomment-1195726642 by @cgeoga  
+# 
+# It is also possible to use continued fraction to calculate inu/inmu1 such as
+# inum1 = besseli_power_series(nu-1, x)
+# H_inu = steed(nu, x)
+# inu = besseli_power_series(nu, x)#inum1 * H_inu
+# but it appears to be faster to modify the series to calculate both Inu and Inum1
+
+function besselk_continued_fraction(nu, x)
+    inu, inum1 = besseli_power_series_inu_inum1(nu, x)
+    H_knu = besselk_ratio_knu_knup1(nu-1, x)
+    return 1 / (x * (inum1 + inu / H_knu))
+end
+
+# a modified version of the I_{nu} power series to compute both I_{nu} and I_{nu-1}
+# use this along with the continued fractions for besselk
+function besseli_power_series_inu_inum1(v, x::T) where T
+    MaxIter = 3000
+    out = zero(T)
+    out2 = zero(T)
+    x2 = x / 2
+    xs = x2^v
+    gmx = xs / gamma(v)
+    a = gmx / v
+    b = gmx / x2
+    t2 = x2 * x2
+    for i in 0:MaxIter
+        out += a
+        out2 += b
+        abs(a) < eps(T) * abs(out) && break
+        a *= inv((v + i + one(T)) * (i + one(T))) * t2
+        b *= inv((v + i) * (i + one(T))) * t2
+    end
+    return out, out2
+end
+
+# computes K_{nu+1}/K_{nu} using continued fractions and the modified Lentz method
+# generally slow to converge for small x
+function besselk_ratio_knu_knup1(v, x::T) where T
+    MaxIter = 1000
+    (hn, Dn, Cn) = (1e-50, zero(v), 1e-50)
+
+    jf = one(T)
+    vv = v * v
+    for _ in 1:MaxIter
+        an = (vv - ((2*jf - 1)^2) * T(0.25))
+        bn = 2 * (x + jf)
+        Cn = an / Cn + bn
+        Dn = inv(muladd(an, Dn, bn))
+        del = Dn * Cn
+        hn *= del
+        abs(del - 1) < eps(T) && break
+        jf += one(T)
+    end
+    xinv = inv(x)
+    return xinv * (v + x + 1/2) + xinv * hn
+end
+
+#####
+#####  Power series for K_{nu}(x)
+#####
+
+# Use power series form of K_v(x) which is accurate for small x (x<2) or when nu > X
+# We use the form as described by Equation 3.2 from reference [7].
+# This method was originally contributed by @cgeoga https://github.com/cgeoga/BesselK.jl/blob/main/src/besk_ser.jl
+# A modified form appears below. See more discussion at https://github.com/heltonmc/Bessels.jl/pull/29
+# This is only accurate for noninteger orders (nu) and no checks are performed. 
+#
+"""
+    besselk_power_series(nu, x::T) where T <: Float64
+
+Computes ``K_{nu}(x)`` using the power series when nu is not an integer.
+In general, this is most accurate for small arguments and when nu > x.
+No checks are performed on nu so this is not accurate when nu is an integer.
+"""
+function besselk_power_series(v, x::T) where T
+    MaxIter = 1000
+    z  = x / 2
+    zz = z * z
+    
+    logz = log(z)
+    xd2_v = exp(v*logz)
+    xd2_nv = inv(xd2_v)
```

Comment:
I'm still unsure if this confuses what is going on here and shouldn't just be the literal power. Considering I think the powers are going to much faster on 1.9....

Reply:
Was actually thinking about this too. Maybe @oscardssmith should make a `BetterPow.jl` in the mean time, even, and we could then use that....

Reply:
It is about 5-10% faster.... but inconsistent with the rest of the power series. Maybe we will leave it for now with a note to look later...